### PR TITLE
 Move added client name IdentityUserAssignedIdentitiesValue into an inner scope

### PR DESCRIPTION
**Pull request by Grayer123**
### Latest improvements:
Related to a [previous pr: bug fix](https://github.com/Azure/azure-rest-api-specs/pull/9272), verified the changes in dotnet SDK, found out that the client name **"IdentityUserAssignedIdentitiesValue"** should be moved into an inner **"additionalProperties"** scope. Re-generated SDK client from local swagger file and verified in dotnet SDK.

Related to this issue: Add **client-name** for **userAssignedIdentities** to avoid discrepant generation between **autorest.dotnet-v2** and **autorest.dotnet-v3**: [autorest.dontnet-v3 generates random name: Componentssgqdofschemasidentitypropertiesuserassignedidentitiesadditionalproperties for userAssignedIdentities](https://github.com/Azure/autorest.csharp/issues/614)
<br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/resources/resource-manager/Microsoft.Resources/stable/2019-10-01/resources.json [hunk 1]
```diff
@@ -5034,9 +5034,9 @@
         },
         "userAssignedIdentities": {
           "type": "object",
-          "x-ms-client-name": "IdentityUserAssignedIdentitiesValue",
           "additionalProperties": {
             "type": "object",
+            "x-ms-client-name": "IdentityUserAssignedIdentitiesValue",
```
Comment (raych1) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2019-10-01/resources.json:5039]: Is this required to define within `additionalProperties`? I don't see it from the [extension document](http://azure.github.io/autorest/extensions/#x-ms-client-name).

Comment (Grayer123) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2019-10-01/resources.json:5039]: The reason why I added **client-name** is that **autorest-v3** does not know how to name this **additionalProperties**. It once generated a random name: **Componentssgqdofschemasidentitypropertiesuserassignedidentitiesadditionalproperties**, which is inconsistent with the name **IdentityUserAssignedIdentitiesValue** autorest-v2 generated. The details can be found here: https://github.com/Azure/autorest.csharp/issues/614
My last pr incorrectly put it in an upper level, and I found this issue in dotnet sdk; 
Moved it to the inner scope, re-verified it worked correctly.

**Comment by raych1** [comment]
@Grayer123 , can you have a look at the [breaking change in python sdk](https://github.com/Azure/azure-rest-api-specs/pull/9363#issuecomment-624904374).

**Comment by Grayer123** [comment]
> @Grayer123 , can you have a look at the [breaking change in python sdk](https://github.com/Azure/azure-rest-api-specs/pull/9363#issuecomment-624904374).

Hi @raych1, I did take a look of the python breaking change:
```
[ChangeLog] **Breaking changes**
[ChangeLog] 
[ChangeLog]   - Model AliasPathType no longer has parameter pattern
```
I think it's related to this pr: https://github.com/Azure/azure-rest-api-specs/pull/9091
It has the label **"potential-sdk-breaking-change"**, which also has the **AliasPathType** related change.

**Review by raych1 — APPROVED** [review]

**Comment by raych1** [comment]
@Tiano2017 , can you please review the change?

**Comment by raych1** [comment]
@Tiano2017 , can you please review the change?

**Review by Tiano2017 — APPROVED** [review]

**Comment by Grayer123** [comment]
Hello @raych1, could you please help us merge it? Thanks

**Pull request merged**